lib/syskit/roby_app/plugin.rb

```diff
@@ -942,6 +957,51 @@ def self.setup_rest_interface(app, rest_api)
                 require "syskit/roby_app/rest_api"
                 rest_api.mount REST_API => "/syskit"
             end
+
+            def get_running_agents(syskit_task)
+                agents = []
+                plan.find_tasks(syskit_task).running.each do |task|
+                    agents << task.execution_agent.arguments[:on]
+                end
+                agents&=agents
+                agents
+            end
+
+            def rotate_logs(rotated_logs)
+                agents = get_running_agents(Syskit::LoggerService)
+                for agent in agents
+                    agent_logs = []
+                    plan.find_tasks(Syskit::LoggerService).running.each do |task|
+                        if task.execution_agent.arguments[:on] == agent
+                            task.rotate_log.each {|logfile| agent_logs << logfile}
+                        end
+                    end
```

Comment:
I don't get the logic here, is there a different plan for each agent iteration?

Also, looks like that `get_running_agents(Syskit::LoggerService)` is basically calling 
```rb
plan.find_tasks(Syskit::LoggerService).running.each
``` 
Why would we have to do it again inside the agents iteration, and why comparison `task.execution_agent.arguments[:on] == agent`? Isn't this always going to be true, or am I missing something?

Reply:
You are absolutely right. It's a very complicated way, I think driven by the Mateus' need to group the logfiles per-process server (the :on argument is the process server name, `agent` is a misnomer. The execution agent is a task's deployment)